packages/pirania/files/usr/libexec/rpcd/pirania

```diff
@@ -19,71 +19,105 @@ local uci = require('uci')
 local uci_cursor = uci.cursor()
 
 local function printJson (obj)
-    print(json.encode(obj))
+    print(json.stringify(obj))
 end
 
 local conn = ubus.connect()
 if not conn then
     error("Failed to connect to ubus")
 end
 
-local function shell(command)
-    -- TODO(nicoechaniz): sanitize or evaluate if this is a security risk
-    local handle = io.popen(command)
-    local result = handle:read("*a")
-    handle:close()
-    return result
+local function status(msg)
+    local output = logic.iptables_status()
+    printJson(output)
+end
+
+local function enable(msg)
+    uci_cursor:set("pirania", "base_config", "enabled", "1")
```

Comment:
is it needed to both set enabled=1 and run captive-portal start?
@spiccinini

Reply:
Setting `enabled` makes pirania start on startup. To make it run you need to `captive-portal start`.

Reply:
Maybe we can use the /etc/init.d/ scripts for the startup?